java/junit5/src/main/java/com/saucelabs/saucebindings/junit5/SauceBindingsExtension.java

```diff
@@ -1,114 +1,171 @@
+        ExtensionContext.Namespace.create(getClass(), context.getRequiredTestMethod()));
   }
 
   @Override
   public void testSuccessful(ExtensionContext context) {
-    if (isExtensionDisabled()) {
-      return;
+    if (!SauceSession.isDisabled()) {
+      SauceSession session = (SauceSession) getStore(context).get("session");
+      RemoteWebDriver driver = session.getDriver();
+      try {
+        session.stop(true);
+      } catch (NoSuchSessionException e) {
+        LOGGER.severe(
+            "Driver quit prematurely; Remove calls to `driver.quit()` to allow SauceBindingsExtension"
+                + " to stop the test");
+      }
     }
+  }
 
-    try {
-      session.stop(true);
-    } catch (NoSuchSessionException e) {
-      logger.severe(
-          "Driver quit prematurely; Remove calls to `driver.quit()` to allow SauceBindingsExtension"
-              + " to stop the test");
+  @Override
+  public void testFailed(ExtensionContext context, Throwable cause) {
+    if (!SauceSession.isDisabled()) {
+      SauceSession session = (SauceSession) getStore(context).get("session");
+      try {
+        session.annotate("Failure Reason: " + cause.getMessage());
+
+        Arrays.stream(cause.getStackTrace())
+            .map(StackTraceElement::toString)
+            .filter(line -> !line.contains("sun"))
+            .forEach(session::annotate);
+
+        session.stop(false);
+      } catch (NoSuchSessionException e) {
+        LOGGER.severe(
+            "Driver quit prematurely; Remove calls to `driver.quit()` to allow SauceBindingsExtension"
+                + " to stop the test");
+      }
     }
   }
 
   @Override
-  public void testFailed(ExtensionContext context, Throwable cause) {
+  public void testAborted(ExtensionContext context, Throwable cause) {
+    LOGGER.fine("Test Aborted: " + cause.getMessage());
+    SauceSession session = (SauceSession) getStore(context).get("session");
     if (session != null) {
-      session.annotate("Failure Reason: " + cause.getMessage());
+      session.annotate("Test Aborted; marking completed instead of failed");
+      session.annotate("Reason: " + cause.getMessage());
 
       Arrays.stream(cause.getStackTrace())
-          .map(StackTraceElement::toString)
-          .filter(line -> !line.contains("sun"))
-          .forEach(session::annotate);
+              .map(StackTraceElement::toString)
+              .filter(line -> !line.contains("sun"))
+              .forEach(session::annotate);
 
-      session.stop(false);
+      session.abort();
     }
   }
 
-  // TODO: Implement this in SauceSession directly
-  private boolean isExtensionDisabled() {
-    String value = System.getenv("SAUCE_DISABLED");
-    return Boolean.parseBoolean(value) || Boolean.getBoolean("sauce.disabled");
+  @Override
+  public void testDisabled(ExtensionContext context, Optional<String> reason) {
+    LOGGER.info("A Sauce session was not started for " + context.getDisplayName() + " because " + reason);
```

Comment:
Not necessarily(?) tests can be disabled dynamically, and it might not be obvious looking at the console log which apply? (see https://github.com/saucelabs/sauce_bindings/pull/334/files/cde48e73218c144a1ed5acc804d063c01fe5f7a1#diff-fb8851e65411919c7b5db33a84c31f6f1ec2554e6ddc6118952da080bc55499eR17)

Whether logging is necessary because the information already shows up in normal maven execution is possible. I think we need to have something implemented there, and I chose to log a statement about how the status affects this extension. If we think it is duplicated, we can set it to FINE instead?

Reply:
So when it isn't disabled, it looks like:
```
[INFO] Running com.saucelabs.saucebindings.junit5.ToggleTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.276 s -- in com.saucelabs.saucebindings.junit5.ToggleTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
``` 

When it is disabled, without the output is:
```
[INFO] Running com.saucelabs.saucebindings.junit5.ToggleTest
Oct 11, 2024 12:43:39 PM com.saucelabs.saucebindings.junit5.SauceBindingsExtension testDisabled
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.075 s -- in com.saucelabs.saucebindings.junit5.ToggleTest
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1
```
It doesn't say what was disabled or why. 
I think there is value in noting that tests were disabled because of that property getting set and which they are:
```
INFO: A Sauce session was not started for disableSauce(SauceSession, WebDriver) because Optional[System property [sauce.disabled] does not exist]
```